### PR TITLE
[13.0][IMP] l10n_eu_oss: tax groups are created and searched for the new taxes.

### PR DESCRIPTION
**Pull request by ValentinVinagre**
tax groups are created and searched for the new taxes.

### CODE REVIEW

File: l10n_eu_oss/wizard/l10n_eu_oss_wizard.py [hunk 1]
```diff
@@ -90,6 +90,9 @@ def _default_todo_country_ids(self):
         comodel_name="account.tax", string="Second Super Reduced Tax"
     )
 
+    def _prepare_tax_group_vals(self, rate):
+        return {"name": _("OSS VAT {}{}".format(rate, "%"))}
```
Comment (pedrobaeza) [inline l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: Use old replacing system (`%s`) and don't include in the translatable part the replacing itself:
```suggestion
        return {"name": _("OSS VAT %s%%") % rate}
```

File: l10n_eu_oss/wizard/l10n_eu_oss_wizard.py [hunk 2]
```diff
@@ -194,8 +199,15 @@ def generate_eu_oss_taxes(self):
                         limit=1,
                     )
                     if not tax_dest_id:
+                        tax_group_id = account_tax_group.search(
+                            [("name", "=", _("OSS VAT {}{}".format(rate, "%")))]
```
Comment (pedrobaeza) [inline l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: Don't recreate the same group name, but call `self._prepare_tax_group_vals()["name"]`

File: l10n_eu_oss/wizard/l10n_eu_oss_wizard.py [hunk 3]
```diff
@@ -194,8 +199,15 @@ def generate_eu_oss_taxes(self):
...
+                        )
+                        if not tax_group_id:
+                            tax_group_id = account_tax_group.create(
+                                self._prepare_tax_group_vals(rate)
+                            )
                         tax_dest_id = account_tax.create(
-                            self._prepare_tax_vals(country, tax, rate)
+                            self._prepare_tax_vals(country, tax, rate, tax_group_id)
```
Comment (pedrobaeza) [inline l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: Don't change the arguments, but do the search/create inside the `_prepare_tax_vals` method IMO

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: l10n_eu_oss/wizard/l10n_eu_oss_wizard.py (hunk 3 above)
Comment (ValentinVinagre) [reply l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: I don't want to search / create the tax group in _prepare_tax_vals, I think this method should only add the fields to the dictionary, how do you see it?

Comment (pedrobaeza) [reply l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: OK, I don't have right now an strong opinion on this. It's for being more self-contained.

File: l10n_eu_oss/wizard/l10n_eu_oss_wizard.py [hunk 4]
```diff
@@ -108,7 +111,7 @@ def _prepare_repartition_line_vals(self, original_rep_lines):
             for line in original_rep_lines
         ]
 
-    def _prepare_tax_vals(self, country_id, tax_id, rate):
+    def _prepare_tax_vals(self, country_id, tax_id, rate, tax_group_id):
```
Comment (pedrobaeza) [inline l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: Please don't use the suffix `_id` when you are passing a recordset.

```suggestion
    def _prepare_tax_vals(self, country_id, tax_id, rate, tax_group):
```

File: l10n_eu_oss/wizard/l10n_eu_oss_wizard.py [hunk 5]
```diff
@@ -194,8 +199,15 @@ def generate_eu_oss_taxes(self):
                         limit=1,
                     )
                     if not tax_dest_id:
+                        tax_group_id = account_tax_group.search(
```
Comment (pedrobaeza) [inline l10n_eu_oss/wizard/l10n_eu_oss_wizard.py]: The same here about suffix.

**Review by pedrobaeza — APPROVED** [review]

**Comment by ValentinVinagre** [comment]
thanks for review @pedrobaeza

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍

**Comment by Jortolsa-S73** [comment]
Today we will publish the PR with the migration of this module to 14.0.

**Comment by ValentinVinagre** [comment]
> Today we will publish the PR with the migration of this module to 14.0.

Perfect @Jortolsa . Can you say it in https://github.com/OCA/account-fiscal-rule/issues/138 ? Thanks

**Pull request merged**